src/Admin/AjaxAction.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Class AjaxAction.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+/**
+ * Base class to define a new AJAX action.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class AjaxAction {
+
+
+	/**
+	 * The AJAX action should be accessible by any visitor.
+	 *
+	 * @var int
+	 */
+	const ACCESS_ANY = 0;
+
+	/**
+	 * The AJAX action should only be accessible by authenticated users.
+	 *
+	 * @var int
+	 */
+	const ACCESS_USER = 1;
+
+	/**
+	 * The AJAX action should only be accessible by unauthenticated visitors.
+	 *
+	 * @var int
+	 */
+	const ACCESS_NO_USER = 2;
+
+	/**
+	 * The AJAX handler listens on both the backend and the frontend.
+	 *
+	 * @var int
+	 */
+	const SCOPE_BOTH = 0;
+
+	/**
+	 * The AJAX handler listens on the backend only.
+	 *
+	 * @var int
+	 */
+	const SCOPE_BACKEND = 1;
+
+	/**
+	 * The AJAX handler listens on the frontend only.
+	 *
+	 * @var int
+	 */
+	const SCOPE_FRONTEND = 2;
```

Comment:
We have no use case for frontend scope, do we? Perhaps the scopes should be omitted in favor of leaving this to be exclusively the realm of the admin? If we add actions to the frontend, I'm sure it will look different. 

I think the scopes can be removed.